### PR TITLE
[Flutter-Parent][MBL-13617] Assignment details

### PR DESCRIPTION
**Pull request by CalvinKern**
This does not include discussion header for discussion assignments. Will do in a separate ticket.

To test:
Look at lots of different assignments, quizzes, (discussions still load the message body, but it doesn't look like a discussion header), submitted, not submitted, locked, etc.........

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1019634&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1019635&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+            Text(L10n(context).assignmentDetailsTitle),
+            if (snapshot.hasData)
+              Text(snapshot.data.courseName ?? '', style: Theme.of(context).primaryTextTheme.caption),
+          ],
+        ),
+      );
+
+  Widget _fab() => FloatingActionButton(
+        onPressed: _sendMessage,
+        child: Semantics(
```
Comment (JordanMarshall) [inline apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart]: Optional change: FloatingActionButton has a `tooltip` property you can set that will provide both semantics and a long-press tooltip.

File: apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    final textTheme = Theme.of(context).textTheme;
+
+    final l10n = L10n(context);
+    final alarm = snapshot.data.alarm;
+    final assignment = snapshot.data.assignment;
+    final submission = assignment.submission(widget.studentId);
+    final showStatus = assignment.isSubmittable() || submission.isGraded();
+    final submitted = submission?.submittedAt != null;
+    final submittedColor = submitted
+        ? ParentTheme.of(context).getColorVariantForCurrentState(StudentColorSet.shamrock)
```
Comment (JordanMarshall) [inline apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart]: Should this be `ParentColors.success`? Or does that end up failing the a11y contrast ratio?

File: apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+      SizedBox(height: 8),
+      child,
+      SizedBox(height: 16),
+    ];
+  }
+
+  List<Widget> _lockedRow(Assignment assignment) {
+    final l10n = L10n(context);
+    String message = null;
+    if (assignment.unlockAt != null) {
```
Comment (JordanMarshall) [inline apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart]: Will we need to look at data in `Assignment.lockInfo` and/or `Assignment.lockedForUser` as well? I don't know if `Assignment.unlockAt`/`Assignment.lockAt` will tell us if an assignment is locked due to, say, being part of a locked module.

File: apps/flutter_parent/lib/l10n/app_localizations.dart [hunk 4]
```diff
@@ -488,6 +493,50 @@ class AppLocalizations {
         args: [studentName],
       );
 
+  /// Assignment Details Screen
+
+  String get assignmentDetailsTitle => Intl.message('Assignment Details');
+
+  String assignmentTotalPoints(String points) => Intl.message('$points pts',
```
[the remark this answers is not in the document]
Comment (CalvinKern) [reply apps/flutter_parent/lib/l10n/app_localizations.dart]: That would be nice. I think I'd prefer just using 'points' everywhere. Hopefully Dave agrees.

File: apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart (hunk 2 above)
Comment (CalvinKern) [reply apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart]: I think we may just want to remove ParentColors.success. I don't think it's used anywhere and it doesn't change with hc mode. The designs matched the shamrock colors for dark/hc mode, which is why I couldn't just use the success color.

File: apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart (hunk 3 above)
Comment (CalvinKern) [reply apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart]: I actually had implemented this, and then realized that observers don't seem to get locked out of things like that. We were at least never getting lock info back on an observer. On web, I could see the description for every assignment, regardless of it's unlock_at or it's module requirements.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart (hunk 2 above)
Comment (JordanMarshall) [reply apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart]: 🆗

File: apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart (hunk 3 above)
Comment (JordanMarshall) [reply apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart]: This one appears to be locked on web for an observer: https://jmarshall.instructure.com/courses/1/assignments/113

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1019766&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart (hunk 3 above)
Comment (JordanMarshall) [reply apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart]: Or if you think it's something we should tackle post-beta we can do that too.

Comment (CalvinKern) [reply apps/flutter_parent/lib/screens/assignments/assignment_details_screen.dart]: I don't have a problem tackling it now. Mostly trying to understand the problem. I'm not seeing this on my account, but I do think we should address this (the work is there too, just stashed).

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1019777&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1019778&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**